lib/matplotlib/_constrained_layout.py

```diff
@@ -202,245 +202,253 @@ def do_constrained_layout(fig, renderer, h_pad, w_pad,
-                alignleft = False
-                alignright = False
-                alignbot = False
-                aligntop = False
-                alignheight = False
-                alignwidth = False
-                for mm in range(nn+1, len(axs)):
-                    axc = axs[mm]
-                    rownumCmin, colnumCmin = rownummin[mm], colnummin[mm]
-                    rownumCmax, colnumCmax = rownummax[mm], colnummax[mm]
-                    widthC, heightC = width[mm], height[mm]
-                    # Horizontally align axes spines if they have the
-                    # same min or max:
-                    if not alignleft and colnum0min == colnumCmin:
-                        # we want the _poslayoutboxes to line up on left
-                        # side of the axes spines...
-                        layoutbox.align([ax._poslayoutbox,
-                                         axc._poslayoutbox],
-                                        'left')
-                        alignleft = True
-
-                    if not alignright and colnum0max == colnumCmax:
-                        # line up right sides of _poslayoutbox
-                        layoutbox.align([ax._poslayoutbox,
-                                         axc._poslayoutbox],
-                                        'right')
-                        alignright = True
-                    # Vertically align axes spines if they have the
-                    # same min or max:
-                    if not aligntop and rownum0min == rownumCmin:
-                        # line up top of _poslayoutbox
-                        _log.debug('rownum0min == rownumCmin')
-                        layoutbox.align([ax._poslayoutbox, axc._poslayoutbox],
-                                        'top')
-                        aligntop = True
-
-                    if not alignbot and rownum0max == rownumCmax:
-                        # line up bottom of _poslayoutbox
-                        _log.debug('rownum0max == rownumCmax')
-                        layoutbox.align([ax._poslayoutbox, axc._poslayoutbox],
-                                        'bottom')
-                        alignbot = True
-                    ###########
-                    # Now we make the widths and heights of position boxes
-                    # similar. (i.e the spine locations)
-                    # This allows vertically stacked subplots to have
-                    # different sizes if they occupy different amounts
-                    # of the gridspec:  i.e.
-                    # gs = gridspec.GridSpec(3,1)
-                    # ax1 = gs[0,:]
-                    # ax2 = gs[1:,:]
-                    # then drows0 = 1, and drowsC = 2, and ax2
-                    # should be at least twice as large as ax1.
-                    # But it can be more than twice as large because
-                    # it needs less room for the labeling.
+    for nnn in range(2):
```

Comment:
Please add at least a comment why this is there.

*General note on code structure:*
The length and nesting level of the function smells liike it would like to be refactored to smaller (private) functions. Then, also the for loop could be replaced with twice a call to a function (e.g. `_calc_margins()` - didn't look at the details so the name is probably not good).

Just an observation. Not necessary a call for action.